### ISSUE TITLE
Declarative JPMS support

### ISSUE THREAD

**Issue by vietj** [webauthn4j/webauthn4j#989]
**Is your feature request related to a problem? Please describe.**

Webauthn4j does not support Java modules. Although it might be possible to add support for named automatic modules, it would be better to support explicit modules (with a module-info.class descriptor). 

**Describe the solution you'd like**

Add module-info declarations describing the modularity of the library.

**Describe alternatives you've considered**

None.

**Additional context**

- Vert.x 5 might use webauthn4j and provides support for JPMS, having explicit modules makes definitely a difference.
- There might be some issues for this task: the Apache Kerby ASN1 parser does not provide module support and  a workable solution should be found
- I can contribute support for this, if needed or help testing the JPMS aspect . I did bring JPMS support to the Netty library (https://github.com/netty/netty/issues/14176)

**Comment by ynojima** [comment]
Automatic module name will be added in [the next release](https://github.com/webauthn4j/webauthn4j/pull/988),
but yes, Apache Kerby ASN1 parser is the issue for making webauthn4j named module.

This could be a good chance to implement its own ASN.1 DER parser using [vertx-auth asn1 parser](https://github.com/eclipse-vertx/vertx-auth/blob/master/vertx-auth-common/src/main/java/io/vertx/ext/auth/impl/asn/ASN1.java) as a reference and remove the dependency on Apache Kerby. I'll look into it.

**Comment by vietj** [comment]
thanks for the responsiveness @ynojima copying the vertx parser is what I meant with "workable solution"

**Comment by ynojima** [comment]
I'm working on replacing ASN.1 parser (https://github.com/webauthn4j/webauthn4j/pull/1004). Please wait a little longer.

**Comment by ynojima** [comment]
@vietj I've made a pull-request to support JPMS: https://github.com/webauthn4j/webauthn4j/pull/1005
I confirmed it works with [a modularized app](https://github.com/webauthn4j/quarkus-security-passkeys-poc/pull/1/files), but this is my first Java module, could you review it meets your requirement?

**Comment by vietj** [comment]
@ynojima I'll have a look

**Comment by vietj** [comment]
I think we can integrate this in vertx-auth-webauthn4j and write test using the module path to check it works correctly

**Comment by vietj** [comment]
good job @ynojima thank you!

**Comment by ynojima** [comment]
Thank you for response!

The artifact is published to `https://oss.sonatype.org/content/repositories/snapshots` repository.
Please use following cordinate when performing the test.
```
        <dependency>
            <groupId>com.webauthn4j</groupId>
            <artifactId>webauthn4j-core-async</artifactId>
            <version>0.28.0-SNAPSHOT</version>
        </dependency>
```

**Comment by ynojima** [comment]
@vietj ,
Now JPMS support is released as `0.28.0.RELEASE`: https://github.com/webauthn4j/webauthn4j/releases/tag/0.28.0.RELEASE
